ecl/hql/hqlexpr.cpp

```diff
@@ -5474,7 +5475,7 @@ void CUsedTablesBuilder::removeParent(IHqlExpression * expr)
         if (!root->hasAttribute(newAtom))
             break;
         expr = root->queryChild(0);
-        removeActive(expr->queryNormalizedSelector());
+        removeActiveSelector(expr->queryNormalizedSelector());
```

Comment:
@ghalliday Why is removeActive used in line 5468 rather than removeActiveSelector?

Reply:
This is covering an unusual case e.g., sum(ds.childds, f()), where removeParent(ds.childds) is called to remove the active cursors for the SUM() operator.

I think the logic is that removeActive(ds) will remove (ds.childs, and ds.childds.grand).  The loop then goes to the containing dataset (ds), and removes anything that begins with ds - since it is implicitly in scope.  There is no need to check for nested items since they have already been removed.

The issue shouldn't occur once the graph has been normalized, but before it is normalized it reduces the number of false-positives.

@shamser Does that make any sense?

Reply:
@ghalliday Yes. Thanks for the explanation.